File: basic/maths/countnumofdig.py
def countnumofdig(n):
    countnum = 0
    if(n==0):
        return 0
    while(n != 0):
        countnum += 1
        n = n//10
    return countnum
n = int(input("enter the value: "))
print(f"the num of digits {n} have : {countnumofdig(n)}")

File: basic/maths/countnumofdig.cpp
#include <iostream>
using namespace std;
int countdig(int n){
    if(n==0) return 0;
    int count = 0;
    while(n!=0){
        count++;
        n/=10;
    }
    return count;
}
int main(){
    int n;
    cout << "enter the value of number: ";
    cin >> n;
    cout << "the num of digits : " << countdig(n);
}